Frameworks/CoreMotion/CMMotionManager.mm

```diff
@@ -249,56 +321,140 @@ - (void)stopGyroUpdates {
         }
         
-        self.gyrometer.reportInterval = static_cast<unsigned int>(self.gyroUpdateInterval * c_intervalScaleFactor);
+        self.gyrometer.reportInterval = static_cast<unsigned int>(self.gyroUpdateInterval * c_secondToMilliseconds);
     }
 }
 
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)startMagnetometerUpdatesToQueue:(NSOperationQueue*)queue withHandler:(CMMagnetometerHandler)handler {
-    UNIMPLEMENTED();
+    
+    if (self.magnetometerAvailable) {
+        [self startMagnetometerUpdates];
+        self.magnetometerQueue = queue;
+
+        self.magnetometerToken = [self.magnetometer addReadingChangedEvent:
+            ^void(WDSMagnetometer* sender, WDSMagnetometerReadingChangedEventArgs* e) {
+
+                WDSMagnetometerReading* reading = e.reading;
+
+                CMMagnetometerData* data = [[CMMagnetometerData alloc] initWithValues:reading.magneticFieldX 
+                                                                                    y:reading.magneticFieldY
+                                                                                    z:reading.magneticFieldZ
+                                                                                 time:[self toSecondsSinceBoot:reading.timestamp]];
+                [queue addOperationWithBlock:^{ handler(data, nil); }];                                   
+            }];          
+    }    
 }
 
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)startMagnetometerUpdates {
-    UNIMPLEMENTED();
+    
+    @synchronized(self) {
+      
+        if (self.magnetometerAvailable) {
+        
+            if (self.magnetometerActive) {
+                [self stopMagnetometerUpdates];
+            }
+        
+            self.magnetometerActive = true;
+
+            self.magnetometer.reportInterval = static_cast<unsigned int>(self.magnetometerUpdateInterval * c_secondToMilliseconds);
+        } 
+    }               
 }
 
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 - (void)stopMagnetometerUpdates {
-    UNIMPLEMENTED();
+    
+    @synchronized(self) {
+
+        if (self.magnetometerActive) {
+            self.magnetometerActive = false;
+        
+            // the reportInterval is set to 0, the default value.
+            self.magnetometer.reportInterval = 0;
+        
+            if (self.magnetometerToken.value != 0) {
+                [self.magnetometer removeReadingChangedEvent:self.magnetometerToken];
+                self.magnetometerToken = {0};
+            }
+        
+            if (self.magnetometerQueue) {
+                [self.magnetometerQueue cancelAllOperations];
+                self.magnetometerQueue = nil;
+            }
+        }
+    }                         
+ }
+ 
+
+ // getter function for on-demand reading from sensor
+ -(CMMagnetometerData*)magnetometerData {
+     
+    WDSMagnetometerReading* reading = [self.magnetometer getCurrentReading];
```

Comment:
If we have code (like in the sample) where the calls are motionManager.magnetometerData.magneticField.x, motionManager.magnetometerData.magneticField.y,motionManager.magnetometerData.magneticField.z, we could end up with three different updates.  Is there value in caching this and the last update time and only call windows if the last update time is longer than the minimum update interval?